project-squirrel-dragon/PresentationLayer/Modules/CardDetailsModule/CardDetailViewController.swift

```diff
@@ -0,0 +1,184 @@
+        configureNameLabel()
+        configureRarityLabel()
+        configureTypeLabel()
+        configurePricesLabel()
+        configureTextField()
+        navigationController?.setNavigationBarHidden(true, animated: animated)
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        navigationController?.setNavigationBarHidden(false, animated: animated)
+    }
+    
+    var callback: ((_ cardId: String?, _ numberOfCollectedCards: Int?) -> Void)?
+
+    // MARK: - Init
+    public func configure(withCard card: CollectableCard) {
+        viewModel.configure(withCard: card)
+    }
+
+    private func configureLabel(_ label: UILabel, withText text: String?) {
+        guard let text = text else {
+            label.isHidden = true
+            return
+        }
+        label.text = text
+    }
+
+    private func configureNationalPokedexNumberLabel() {
+        configureLabel(nationalPokedexNumberLabel,
+                       withText: viewModel.nationalPokedexNumberLabel)
+    }
+
+    private func configureCardImageView() {
+        guard let url = viewModel.cardImageUrl else {
+            return
+        }
+        cardImageView.load(url: url)
+    }
+
+    private func configureNumberLabel() {
+        configureLabel(numberLabel,
+                       withText: viewModel.number)
+    }
+
+    private func configureNameLabel() {
+        configureLabel(nameLabel,
+                       withText: viewModel.pokemonName)
+    }
+
+    private func configureRarityLabel() {
+        configureLabel(rarityLabel,
+                       withText: viewModel.rarity)
+    }
+
+    private func configureTypeLabel() {
+        configureLabel(typeLabel,
+                       withText: viewModel.type)
+    }
+
+    private func configurePricesLabel() {
+        configureLabel(pricesLabel,
+                       withText: viewModel.prices.0)
+        configureLabel(pricesNumberLabel,
+                       withText: viewModel.prices.1)
+    }
+
+    private func configureTextField() {
+        collectedNumberTextField.text = "\(viewModel.totalCardsCollected)"
+    }
+    // MARK: - Gestures
+    @IBAction func addButtonPressed(_ sender: UIButton) {
+        viewModel.addOneCard()
+        configureTextField()
+    }
+
+    @IBAction func minusButtonPressed(_ sender: UIButton) {
+        viewModel.removeOneCard()
+        configureTextField()
+    }
+    
+    @IBAction func doneButtonPressed(_ sender: UIButton) {
+        callback?(viewModel.cardId, viewModel.finalCardsCollected)
+    }
+
+    @IBAction func cancelButtonPressed(_ sender: UIButton) {
+        callback?(nil, nil)
+    }
+}
+
+extension CardDetailViewController: UITextFieldDelegate {
+    func textFieldDidBeginEditing(_ textField: UITextField) {
+        var textField = UITextField()
+
+        let alert = UIAlertController(title: "How many have you collected?", message: "", preferredStyle: .alert)
```

Comment:
Think about maybe adding this functionality to show and alert along with adding `UIAlertActions` to an extension, that will make the work here a little bit neater.